src/test/scala/edu/ie3/simona/model/participant2/StorageModelSpec.scala

```diff
@@ -72,68 +81,134 @@ class StorageModelSpec extends UnitSpec with Matchers {
 
   def buildStorageModel(
       targetSoc: Option[Double] = Option.empty
-  ): StorageModel = {
+  ): StorageModel =
     StorageModel.apply(
       inputModel,
-      1,
-      TimeUtil.withDefaults.toZonedDateTime("2020-01-01T00:00:00Z"),
-      TimeUtil.withDefaults.toZonedDateTime("2020-01-01T01:00:00Z"),
-      0d,
-      targetSoc,
+      StorageRuntimeConfig(targetSoc = targetSoc),
     )
-  }
 
   "StorageModel" should {
+
+    "Determine the current state" in {
+      val storageModel = buildStorageModel()
+
+      val lastTick = 3600L
+
+      val testCases = Table(
+        ("lastEnergy", "power", "duration", "expEnergy"),
+        /* empty storage */
+        // zero power
+        (0.0, 0.0, 3600, 0.0),
+        // zero duration
+        (0.0, 5.0, 0, 0.0),
+        // charging a tiny bit
+        (0.0, 1.0, 1, 0.00025),
+        // charging until half
+        (0.0, 10.0, 20000, 50.0),
+        // charging until almost full
+        (0.0, 10.0, 39999, 99.9975),
+        // charging until full
+        (0.0, 10.0, 40000, 100.0),
+        // overcharging a tiny bit
+        (0.0, 10.0, 40001, 100.0),
+        // discharging
+        (0.0, -10.0, 3600, 0.0),
```

Comment:
Does it make sense to check for impossible operationPoints somewhere within the model?

Reply:
What kind of operating points are you thinking of? Energy above capacity, charging power above max power?

Reply:
My main thought here was discharging when empty, charging when full. I'm wondering if this (discharging when empty) should lead to a valid result and not throw some exception. On the other hand it's not good always throw exceptions, as this would be more like an extra safety measure for something that went wrong somewhere else before. I'm not sure what is our general philosophy is on this.

Reply:
Right, it's more a general question of how we check to make sure that this power isn't withdrawn since its not there.

Reply:
That's the responsibility of flex options determination (when EM-controlled) and operating point determination (when not EM-controlled, which is not applicable in this case).